tests/Model/AgentTest.php

```diff
@@ -33,4 +41,131 @@ public function testCanAccess() {
         $this->assertTrue($agent->can_access(array($access)));
         $this->assertFalse($agent->can_access(array($access_bad)));
     }
+
+    public function testIsOnVacationOn(){
+        global $entityManager;
+        $builder = new FixtureBuilder();
+        $builder->delete(Agent::class);
+        $agent = $builder->build(Agent::class, array('login' => 'jdevoe'));
+
+        $start = \DateTime::createFromFormat("d/m/Y", '08/10/2022');
+        $end = \DateTime::createFromFormat("d/m/Y", '15/10/2022');
+
+
+        $builder->delete(Holiday::class);
+        $conge = $builder->build(Holiday::class, array('debut' => $start, 'fin' => $end, 'perso_id' => $agent->id(), 'valide_n1' => 1, 'valide' =>1, 'supprime' => 0, 'information' => 0));
+
+        $this->assertFalse($agent->isOnVacationOn('2022-12-10', '2022-12-15'));
+        $this->assertTrue($agent->isOnVacationOn($start->format('Y-m-d'), $end->format('Y-m-d')));
```

Comment:
Can you add a test with:
  - starts on 2022-10-10, ends on 2022-10-11,
  - starts on 2022-10-14, ends on 2022-10-16,
  - starts on 2022-10-06, ends on 2022-10-09
  - starts on 2022-10-14, ends on 2022-10-07,
  - empty start and/or end variables